llvm/lib/Target/NVPTX/NVPTXTargetTransformInfo.cpp

```diff
@@ -290,15 +290,16 @@ static Instruction *simplifyNvvmIntrinsic(IntrinsicInst *II, InstCombiner &IC) {
     case Intrinsic::nvvm_d2ull_rz:
     case Intrinsic::nvvm_f2ull_rz:
       return {Instruction::FPToUI};
-    case Intrinsic::nvvm_i2d_rz:
-    case Intrinsic::nvvm_i2f_rz:
-    case Intrinsic::nvvm_ll2d_rz:
-    case Intrinsic::nvvm_ll2f_rz:
+    // Integer to floating-point uses RN rounding, not RZ
+    case Intrinsic::nvvm_i2d_rn:
```

Comment:
according to [LLVM IR spec](https://llvm.org/docs/LangRef.html#fptoui-to-instruction) fptoui should be rounding toward zero:
```
The ‘fptoui’ instruction converts its [floating-point](https://llvm.org/docs/LangRef.html#t-floating) operand into the nearest (rounding towards zero) unsigned integer value.
```

Reply:
`fptoui` has nothing to do with this we're converting into-to-FP here and LLVM IR does specify using the default rounding mode for both `uitofp` and `sitofp`. https://llvm.org/docs/LangRef.html#uitofp-to-instruction

Reply:
oh right my bad I got mixed up with line above